src/misc/scanner.ts

```diff
@@ -11,11 +17,13 @@ export const keepScanning = async (
   {
     detectHandler,
     locateHandler,
+    formats = ['qr_code'],
     minDelay
-  }: { detectHandler: any; locateHandler: any; minDelay: any }
+  }: { detectHandler: any; locateHandler: any; minDelay: any, formats: BarcodeFormat[] }
 ) => {
-  const barcodeDetector = new BarcodeDetector({ formats: ['qr_code'] })
-
+  if(!barcodeDetector) {
+    setScanningFormats(formats)
+  }
```

Comment:
Here we might have the situation that the BarcodeDetector instance is preserved over the lifetime of the QrcodeStream component. For example what if we create:
```
<qrcode-stream :formats="['qr_code']" />
```
Do some scanning, destroy the component, and then create a new 
```
<qrcode-stream :formats="['code_93']" />
```
Then I think the BarcodeDetector instance is still alive and we don't create a new one, so we keep scanning with formats `['qr_code']`.

Reply:
sounds good :+1: